datadog.go

```diff
@@ -0,0 +1,277 @@
+package datadog
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// The maximum content size per request is 5MB
+	maxContentSize = 5 * 1024 * 1024
+
+	// The maximum size for a single log is 256kB
+	maxLogSize = 256 * 1024
+
+	// The maximum amount of logs that can be sent in a single request is 1000
+	maxLogCount = 1000
+
+	basePath          = "/v1/input"
+	datadogHostUS     = "https://http-intake.logs.datadoghq.com"
+	datadogHostEU     = "https://http-intake.logs.datadoghq.eu"
+	apiKeyHeader      = "DD-API-KEY"
+	defaultMaxRetries = 5
+)
+
+var (
+	defaultLevel = logrus.InfoLevel
+)
+
+// These options if provided will be added as default tags to each log sent
+type GlobalTags struct {
```

Comment:
These tags are reasonably GFS specific - are we sure it makes sense to include them built into this package?

Would prefer to just have a list of global tags that can be configured by the user.

Reply:
So `Host`, `Service` and `Source` (not here but set statically to `golang` below) are datadog specific, so perhaps there should be added to the `Options` those 3, defaulted if nil, and a `map[string]string` for global tags?

Reply:
Updated